### DISCUSSION TITLE
C API: Support get the 'BIT' length

### DISCUSSION THREAD

**Discussion by wyhaya** [duckdb/duckdb#12615] [Ideas]
### Example
```rs
 let query = conn.query("SELECT '011111'::BIT;")...
// Result:
// bytes length: 2 = 16 bits
// to bits: 0000001011011111

// I'm expecting `011111`, but there is no way to get the length of the bits.
```

It would be great if duckdb had an API for this.

```rs
let length = ffi::duckdb_vector_get_bit_length(duckdb_vector, index)
// length = 6
```

Ultimately, duckdb will allow to get results like those in duckdb-cli through the C API.
```
Connected to a transient in-memory database.
D SELECT '011111'::BIT;
┌───────────────────────┐
│ CAST('011111' AS BIT) │
│          bit          │
├───────────────────────┤
│ 011111                │
└───────────────────────┘
```

**Comment by Giorgi** [comment]
You can already read the bit length and data with C Api. Once you get data from the vector, read it as `duckdb_string_t` and use the code from https://github.com/duckdb/duckdb/blob/7b276746f772e036474e80f19195a0f5af8891a0/src/common/types/bit.cpp#L74 to decode it as BIT

**Reply by Tishj** [reply]
I think upstreaming a method to the C-API that wraps this would be more useful
